### ISSUE TITLE
Generation error with array of enum

### ISSUE THREAD

**Issue by lucasvaudey** [medz/prisma-dart#329]
Hi I run into an error when running `bun prisma generate`. The generated code contains an error :

Schema :
```prisma
model Club {
  id        String       @id @db.Uuid
  name      String
  city      String
  sports    Sport[]
}

enum Sport {
  FOOTBALL
  TENNIS
}
```

Generated model : 
```dart
 factory Club.fromJson(Map json) => Club(
        id: json['id'],
        name: json['name'],
        city: json['city'],
        sports: json['sports'] != null
            ? Iterable<_i2.Sport> //<-- The error is here
                .values
                .firstWhere((e) => e.name == json['sports'])
            : null,
```
And error in `prisma.dart` : 
```dart
factory ClubGroupByOutputType.fromJson(Map json) => ClubGroupByOutputType(
        id: json['id'],
        name: json['name'],
        city: json['city'],
        sports: json['sports'] != null
            ? Iterable<_i3.Sport> //<-- Error is here
                .values
                .firstWhere((e) => e.name == json['sports'])
            : null,
```

Error in code : `The class 'Iterable' doesn't have a constructor named 'values'.
Try invoking a different constructor, or defining a constructor named 'values'`

Version used : `4.0.0-beta.2`
Version of prisma : `^5.8.1`

I fix this generation by doing :

```dart
_i3.Sport
                .values
                .where((e) => e.name == json['sports'])
```

**Comment by NeroSong** [comment]
Hi, according to prisma's doc, I believe your `sports` in Schema is a [Relation](https://www.prisma.io/docs/orm/prisma-schema/data-model/relations).

```
model Club {
  id        String       @id @db.Uuid
  name      String
  city      String
  sports    Sport[]     // this line
}
```

Maybe it would be better to put it in another model and connect them just like doc.

**Comment by medz** [comment]
I think this is an error of the generator. Maybe I didn't consider the situation of many enum fields when setting up enim. I'll check it out, but I'm in the hospital now, so it may take me six or seven hours to repair it.

**Comment by medz** [comment]
https://github.com/medz/prisma-dart/releases/tag/4.0.0-beta.3

Thanks for the feedback, 🎉It's now fixed in `4.0.0-beta.3` version!

**Comment by medz** [comment]
In PGSQL, storing enum arrays is a legal value. While this use case is rare, it is happening now. I'm glad you brought it to my attention so I could fix it.
